app-modules/archeo/src/Services/ActivityExcelParser.php

```
@@ -0,0 +1,267 @@
+                    $errors[] = "Row {$rowIndex} (Activity: {$id}): ".$e->getMessage();
+                }
+            }
+        });
+
+        return [
+            'created' => $createdCount,
+            'updated' => $updatedCount,
+            'count' => $createdCount + $updatedCount,
+            'errors' => $errors,
+        ];
+    }
```

PHPMD (2.15.0)

[warning] The method importFromPath() has a Cyclomatic Complexity of 14. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)